File: armorpaint/Sources/slot_font.ts

class slot_font_t {
	image: image_t = null; // 200px
	preview_ready: bool = false;
	id: i32 = 0;
	font: g2_font_t;
	name: string;
	file: string;
}

function slot_font_create(name: string, font: g2_font_t, file = ""): slot_font_t {
	let raw: slot_font_t = new slot_font_t();
	for (let slot of project_fonts) {
		if (slot.id >= raw.id) {
			raw.id = slot.id + 1;
		}
	}
	raw.name = name;
	raw.font = font;
	raw.file = file;
	return raw;
}

File: paint/sources/slot_font.c
slot_font_t *slot_font_create(char *name, draw_font_t *font, char *file) {
	slot_font_t *raw   = GC_ALLOC_INIT(slot_font_t, {0});
	raw->preview_ready = false;
	raw->id            = 0;

	for (i32 i = 0; i < project_fonts->length; ++i) {
		slot_font_t *slot = project_fonts->buffer[i];
		if (slot->id >= raw->id) {
			raw->id = slot->id + 1;
		}
	}

	raw->name = string_copy(name);
	raw->font = font;
	raw->file = string_copy(file);

	return raw;
}
